[Alqasim Elyas Zuhair A] iP

Eru111 opened the pull request:
# Bruno
> Will store your tasks in my trusty zippers with my *Stickyyyy Fingaaaz* (Sticky Fingers)!

Bruno will zip open your brain, find and store any task you have in mind, and make you **headache-free** :100:

## Features
1. Add and delete tasks
2. Mark and unmark tasks
3. List tasks
4. Store tasks in a data file for future access
5. Find tasks with a keyword

## Commands
- todo `task description`
- deadline `task description` /by `due date (yyyy-MM-dd HH:mm)`
- event `task description` /from `start date (yyyy-MM-dd HH:mm)` /to `end date (yyyy-MM-dd HH:mm)`
- list
- delete `task indices` (e.g. delete 1 2 3)
- mark `task indices` (e.g. mark 1 2 3)
- unmark `task indices` (e.g. unmark 1 2 3)
- find `keyword`
- bye

## What is next?
- [x] Managing tasks
- [x] Saving tasks to disk
- [x] Create GUI

## How is input handled?
If you are interested, here is how the user input is handled using the handleUserInput method:
```
     @FXML
    private void handleUserInput() {
        String input = userInput.getText();
        String response = bruno.getResponse(userInput.getText());
        String commandType = bruno.getCommandType();
        dialogContainer.getChildren().addAll(
                DialogBox.getUserDialog(input, userImage),
                DialogBox.getBrunoDialog(response, brunoImage, commandType)
        );
        userInput.clear();
    }
```
You can access the code [here](https://github.com/Eru111/ip).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename Duke.java to Bruno.java and implement skeletal version

Commit:
Update Bruno to echo and exit on command

Commit:
Update Bruno to add and list

Commit:
Add Task.java and update Bruno to do and undo tasks

Commit:
Add ToDo.java, Deadline.java, Event.java, support for different tasks

Commit:
Done automated text UI testing

Commit:
Add exception handling and test cases for errors

Commit:
Add support for deletion of tasks

Commit:
Use enum for task type

Commit:
Add save to text file feature

Commit:
Handle case where data file does not exist

Commit:
Delete white space

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Bruno.java

Commit:
Change date and time from String to LocalDateTime

Commit:
Remove white space

Commit:
Merge branch 'branch-Level-8'

Commit:
Change implementation to more OOP

Commit:
Add white space

Commit:
Merge branch 'branch-A-MoreOOP'

# Conflicts:
#	src/main/java/Bruno.java

Commit:
Divide java files into multiple packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Add Gradle support to project

Commit:
Add JUnit tests for Storage and Parser classes

Commit:
Edit build.gradle to create jar file

Commit:
Add JavaDocs to methods and classes

Commit:
Add white space

Commit:
Tweak to comply with coding standard

Commit:
Add find task by keyword feature

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/bruno/Ui.java
#	src/main/java/bruno/command/AddCommand.java
#	src/main/java/bruno/command/Command.java
#	src/main/java/bruno/command/DeleteCommand.java
#	src/main/java/bruno/command/ExitCommand.java
#	src/main/java/bruno/command/ListCommand.java
#	src/main/java/bruno/command/MarkCommand.java
#	src/main/java/bruno/command/UnmarkCommand.java
#	src/main/java/bruno/exceptions/ParsingException.java
#	src/main/java/bruno/task/TaskList.java

Commit:
Change to better follow coding standard

Commit:
Remove some file

Commit:
Add checkstyle.xml file

Commit:
Put checkstyle.xml code

Commit:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Put checkstyle.xml code
  Add checkstyle.xml file

Commit:
fix some checkstyle errors

Commit:
Fix all checkstyle errors

Commit:
Create GUI

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Create GUI

Commit:
Use varargs feature

Commit:
Remove unwanted files

maertan commented on src/main/java/bruno/DialogBox.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    private void changeDialogStyle(String commandType) {
+        switch (commandType) {
```
Good job adhering to the coding standards for switch cases!

maertan commented on src/main/java/bruno/Storage.java:
```diff
@@ -0,0 +1,138 @@
+ * and update the file with the current task list.
+ */
+public class Storage {
+    private DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern("MMM d yyyy HH:mm");
```
Perhaps the variable name could be `formatter` or `dateTimeFormatter`? Since there is no other formatter within this file.

maertan commented on src/main/java/bruno/task/TaskList.java:
```diff
@@ -0,0 +1,197 @@
+ */
+public class TaskList {
+    private DateTimeFormatter formatter1 = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    private ArrayList<Task> tasks;
```
Perhaps you could standardise the naming of `tasks`? There was a task list earlier that was named `taskList`.

maertan reviewed (commented):
Overall, I liked how you used OOP and followed the coding standard! Some small suggestions in the comments below.

Jayden-Kim-NUS commented on src/main/java/bruno/command/Command.java:
```diff
@@ -0,0 +1,58 @@
+     * the specific behavior of each command.
+     */
+    public abstract void execute() throws BrunoException;
+    @Override
```
Do consider leaving blank line between these 2 methods

Jayden-Kim-NUS commented on src/main/java/bruno/Storage.java:
```diff
@@ -0,0 +1,138 @@
+
+                String[] lineParts = line.split("\\|");
+                String type = lineParts[0].trim();
+                boolean done = lineParts[1].trim().charAt(1) == 'X';
```
Perhaps consider naming the boolean isDone or similar

Jayden-Kim-NUS reviewed (commented):
Great use of OOP! Code is easy to follow and clean. LGTM.

Commit:
Implement some suggestions from PR comments

Commit:
Add assertions to parts of the code

Commit:
Implement functionality using Java streams

In the deleteTask method, the String array created from the varargs
parameter nums is converted into a stream. Each String in the stream
is then mapped to its parsed integer equivalent, sorted in descending
order, and converted into a list of integers.

Eru111 committed:
Merge pull request #1 from Eru111/branch-A-Streams

Implement functionality using Java streams

Commit:
Merge branch 'master' of https://github.com/Eru111/ip

* 'master' of https://github.com/Eru111/ip:
  Implement functionality using Java streams

Eru111 committed:
Merge pull request #2 from Eru111/branch-A-Assertions

Branch A Assertions

Eru111 committed:
Revert "Branch A Assertions"

Eru111 committed:
Merge pull request #3 from Eru111/revert-2-branch-A-Assertions

Revert "Branch A Assertions"

Eru111 committed:
Revert "Implement functionality using Java streams"

Commit:
Add assertions to parts of the code

Commit:
Improve code quality by following textbook rules

Commit:
Implement functionality using Java streams

In the deleteTask method, the array created by the varargs
parameter nums is converted into a stream. Each string in
the stream is mapped to its parsed integer equivalent, sorted
in reverse order, and converted into a list of integers.

Eru111 committed:
Merge pull request #5 from Eru111/branch-A-Assertions

Add assertions to parts of the code

Commit:
Merge branch 'master' of https://github.com/Eru111/ip

* 'master' of https://github.com/Eru111/ip:
  Add assertions to parts of the code

Commit:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions to parts of the code

Commit:
Merge branch 'master' into branch-A-Streams

* master:
  Add assertions to parts of the code

Eru111 committed:
Merge pull request #6 from Eru111/branch-A-Streams

Branch A Streams

Commit:
Merge branch 'master' of https://github.com/Eru111/ip

* 'master' of https://github.com/Eru111/ip:
  Implement functionality using Java streams

Commit:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Implement functionality using Java streams

# Conflicts:
#	src/main/java/bruno/task/TaskList.java

Eru111 committed:
Merge pull request #7 from Eru111/branch-A-CodeQuality

Branch A CodeQuality

Commit:
Merge branch 'master' of https://github.com/Eru111/ip

* 'master' of https://github.com/Eru111/ip:
  Improve code quality by following textbook rules

Eru111 committed:
Create gradle.yml

Commit:
Merge branch 'master' of https://github.com/Eru111/ip

* 'master' of https://github.com/Eru111/ip:
  Create gradle.yml

Commit:
Add parsing natural dates functionality

Parser class now has multiple methods for dealing
with different date formats entered by the user including
natural dates like "Mon", "today 1am", "tomorrow 13:30",
etc... This makes entering deadlines easier while still
keeping the user's ability to input a farther date with the
yyyy-MM-dd format.

Commit:
Improve GUI and make it fit the bot theme

Commit:
Give the bot a unique personality

Commit:
Add Ui picture to docs folder

Commit:
Add user guide documentation

Commit:
Remove README for Duke project

Commit:
Fix README formatting error
